langkit/toxicity.py

```diff
@@ -14,47 +19,60 @@
 
 _prompt = prompt_column
 _response = response_column
-_toxicity_tokenizer = None
-_toxicity_pipeline = None
 
+_model_path: Optional[str] = None
 
-def toxicity(text: str) -> float:
-    if _toxicity_pipeline is None or _toxicity_tokenizer is None:
-        raise ValueError("toxicity score must initialize the pipeline first")
+_model = LazyInit(
+    lambda: AutoModelForSequenceClassification.from_pretrained(_model_path)
+)
+_toxicity_tokenizer = LazyInit(lambda: AutoTokenizer.from_pretrained(_model_path))
+__use_cuda = torch.cuda.is_available() and not bool(
+    os.environ.get("LANGKIT_NO_CUDA", False)
+)
+_toxicity_pipeline = LazyInit(
+    lambda: TextClassificationPipeline(
+        model=_model.value,
+        tokenizer=_toxicity_tokenizer.value,
+        device=0 if __use_cuda else -1,
+    )
+)
 
-    result = _toxicity_pipeline(
-        text, truncation=True, max_length=_toxicity_tokenizer.model_max_length
+
+def toxicity(text: str) -> float:
+    toxicity_pipeline = _toxicity_pipeline.value
+    toxicity_tokenizer = _toxicity_tokenizer.value
+    result = toxicity_pipeline(
+        text, truncation=True, max_length=toxicity_tokenizer.model_max_length
     )
     return (
         result[0]["score"] if result[0]["label"] == "toxic" else 1 - result[0]["score"]
     )
 
 
+def init_model():
+    if _model_path is None:
+        raise ValueError("Must initialize model path before calling toxicity!")
+    _model.value
+    _toxicity_tokenizer.value
+    _toxicity_pipeline.value
```

Comment:
Curios if this is noticeably slower or not?

Reply:
I did a very simple test, iterating on 5k samples (single row extraction), and `init_model()` takes, on average 0.0048 ms (4.8e-6 seconds).